cloudify_hostpool_plugin/tasks.py

```diff
@@ -15,79 +19,113 @@
 
 import os
 import requests
+from requests.exceptions import RequestException, Timeout
 import stat
 
 from cloudify import ctx
-from cloudify.exceptions import NonRecoverableError
+from cloudify.exceptions import NonRecoverableError, RecoverableError
 from cloudify.decorators import operation
 
 
 RUNTIME_PROPERTIES_KEYS = [
-    'ip', 'user', 'port', 'password', 'key', 'host_id', 'public_address'
+    'ip', 'user', 'port',
+    'password', 'key', 'host_id',
+    'public_address'
 ]
 
 
 @operation
-def acquire(service_url, **kwargs):
+def acquire(service_url, **_):
+    '''Acquire a host for the user'''
     ctx.logger.info('Acquire host')
-    response = requests.post('{0}/hosts'.format(service_url))
-    ctx.logger.debug('Response received: {0}'.format(response.text))
-    if response.status_code == requests.codes.CREATED:
-        host = response.json()
-        ctx.logger.info('Acquired host: {0}'.format(host['host']))
-        key_content = host['auth'].get('keyfile')
-        key_path = None
-        if key_content:
-            key_path = _save_keyfile(key_content, host['host_id'])
-        _set_runtime_properties(host, key_path)
-    else:
-        _handle_error(response)
+    try:
+        response = requests.post('%s/hosts' % service_url)
+        ctx.logger.debug('Response received: %s', response.text)
+
+        # pylint: disable=E1101
+        # Harmless warning due to the dynamically-generated LookupDict
+        if response.status_code == requests.codes.CREATED:
```

Comment:
I think it would make sense to put everything from:
```
if response.status_code == requests.codes.CREATED:
```
to
```
_handle_error(response)
```
into its own function.